src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/SearchObject.cs

```diff
@@ -233,6 +235,51 @@ private void CacheBackgroundData(IPackageSearchMetadata packageSearchMetadata, b
             {
                 memoryCacheItem.UpdateSearchMetadata(packageSearchMetadata);
             }
+
+            Assumes.NotNull(packageSearchMetadata);
+            NuGetRemoteFileService.AddIconToCache(packageSearchMetadata.Identity, packageSearchMetadata.IconUrl);
+
+            string? packagePath = (packageSearchMetadata as LocalPackageSearchMetadata)?.PackagePath ??
+                    (packageSearchMetadata as ClonedPackageSearchMetadata)?.PackagePath;
+
+            if (packagePath != null)
+            {
+                LicenseMetadata? licenseMetadata = (packageSearchMetadata as LocalPackageSearchMetadata)?.LicenseMetadata ??
```

Comment:
Is this giving preference to Local metadata over remote metadata, and is that desirable? I suppose it doesn't matter if "every version has unique metadata", but that's not enforced, right?

Reply:
this is consistent with the approach that the old code had in PackageReader property here: https://github.com/NuGet/NuGet.Client/pull/3823/files#diff-824e76a552194f56f041b295560d22ab96a008470cd002335c12210a9c1d268dR110